lib/chef/application/client.rb

```diff
@@ -383,7 +335,122 @@ def run_application
     end
   end
 
+  def chef_client_json
+    json = Chef::Config.json_attribs
+
+    @chef_client_json ||= Chef::ConfigFetcher.new(json).fetch_json if json
+  end
+
   private
+
+  def verify_audit_mode
+    if %i(enabled audit_only).include?(audit_mode)
+      # TODO: remove this when audit-mode is enabled by default/no longer
+      # an experimental feature.
+      Chef::Log.warn(audit_mode_experimental_message)
+    else
+      handle_unrecognized_audit_mode
+    end
+  end
+
+  def verify_forked_interval
+    Chef::Application.fatal!(unforked_interval_error_message) if
+      !Chef::Config.client_fork && Chef::Config.interval &&
+      !Chef::Platform.windows?
+  end
+
+  def update_interval_and_splay
+    if Chef::Config.daemonize
+      Chef::Config.interval ||= 1800
+    elsif Chef::Config.once
+      Chef::Config.interval = nil
+      Chef::Config.splay = nil
+    end
```

Comment:
Nope, wasn't intentional. 

Looking at the original logic: shouldn't `daemonize` and `once` cause a fatal conflict, like `recipe-url` without `local-mode`?